backend/src/main/java/com/google/minicrm/servlets/FormsServlet.java

```diff
@@ -103,8 +96,12 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response)
       response.sendError(401, "Log in with Google to continue."); //401 Unauthorized
       return;
     }
-
-    long[] formIds = convertLongArrToStringArr(request.getParameterValues("formIds[]"));
+    String[] strFormIds = request.getParameterValues("formIds[]");
+    if (strFormIds == null || strFormIds.length == 0) {
+      response.sendError(400, "formIds[] not specified or is empty.");
+      return;
+    }
+    long[] formIds = convertLongArrToStringArr(strFormIds);
```

Comment:
might be simpler to use a stream here to convert from string to long. I think Arrays.stream() should work, but there may be a better way. For example, Arrays.stream(strFormIds).map(Long::parseLong).collect(toList()); It can also be made into a primitive array I think with .toArray() instead of .collect(), but I'm not 100% sure.

Reply:
Changed to a stream.
Just a quick question: what's the benefit of using a stream over a for loop?

Reply:
not an explicit performance benefit, but it is nice to use streams for readable and purely functional transformations.

Reply:
Gotcha makes sense! thanks